src/wpool.erl

```diff
@@ -22,10 +22,21 @@
                   , {workers, 100}, {worker, {wpool_worker, undefined}}
                   ]).
 
--type name() :: atom().
+-type gen_server_dbg() :: trace
+                        | log
+                        | statistics
+                        | {log_to_file,FileName::term()}
+                        | {install,{Func::term(),FuncState::term()}}.
+
+-type gen_server_dbgs() ::  [gen_server_dbg()].
+-type gen_server_spawn_opts() :: [term()].
+-type gen_server_opts() :: {debug, gen_server_dbgs()}
+                         | {timeout,infinity|pos_integer()}
+                         | {spawn_opt, gen_server_spawn_opts()}.
 -type option() :: {overrun_warning, infinity|pos_integer()}
                 | {overrun_handler, {Module::atom(), Fun::atom()}}
                 | {workers, pos_integer()}
+                | {worker_opt, gen_server_opts()}
```

Comment:
I think you can just use `gen:options()` instead of defining your own new type here.